mundane-io/src/main/scala/com/ambiata/mundane/io/LocalFile.scala

```diff
@@ -0,0 +1,245 @@
+    RIO.using(path.toOverwriteOutputStream)(out => Streams.writeWithEncoding(out, content, encoding))
+
+  def overwriteLines(content: List[String]): RIO[Unit] =
+    overwriteLinesWithEncoding(content, Codec.UTF8)
+
+  def overwriteLinesWithEncoding(content: List[String], encoding: Codec): RIO[Unit] =
+    overwriteWithEncoding(Lists.prepareForFile(content), encoding)
+
+  def overwriteBytes(content: Array[Byte]): RIO[Unit] =
+    RIO.using(path.toOverwriteOutputStream)(Streams.writeBytes(_, content))
+
+  def move(destination: LocalPath): RIO[LocalFile] =
+    doesExist(s"Source file does not exist. LocalFile($path)",
+      destination.doesNotExist(s"A file/directory exists in target location $destination. Can not move source file LocalFile($path)",
+        RIO.safe {
+          val destFile = destination.toFile
+          destination.dirname.toFile.mkdirs
+          this.toFile.renameTo(destFile)
+        }.as(LocalFile.unsafe(destination.path.path))))
+
+  def moveWithMode(destination: LocalPath, mode: TargetMode): RIO[LocalFile] =
+    mode.fold(doesExist(s"Source file does not exist. LocalFile($path)",
+      RIO.safe {
+        val destFile = destination.toFile
+        destination.dirname.toFile.mkdirs
+        this.toFile.renameTo(destFile)
+      }.as(LocalFile.unsafe(destination.path.path))), move(destination))
+
+  def moveTo(destination: LocalDirectory): RIO[LocalFile] =
+    (path.basename match {
+      case None =>
+        RIO.fail(s"Source is a top level directory, can't move. Source($path)")
+      case Some(filename) =>
+        move(destination.toLocalPath | filename)
+    })
+
+  def copy(destination: LocalPath): RIO[LocalFile] =
+    doesExist(s"Source file does not exist. LocalFile($path)",
+      destination.doesNotExist(s"A file/directory exists in target location $destination. Can not move source file LocalFile($path)",
+        destination.dirname.mkdirs >>
+          RIO.using(RIO.safe[InputStream](new FileInputStream(toFile)))(destination.writeStream(_))).as(LocalFile.unsafe(destination.path.path)))
+
+  def copyWithMode(destination: LocalPath, mode: TargetMode): RIO[LocalFile] =
+    mode.fold(doesExist(s"Source file does not exist. LocalFile($path)",
+      destination.dirname.mkdirs >>
+        RIO.using(RIO.safe[InputStream](new FileInputStream(toFile)))(destination.overwriteStream(_))).as(LocalFile.unsafe(destination.path.path)),
+      copy(destination))
+
+  def copyTo(destination: LocalDirectory): RIO[LocalFile] =
+    path.basename match {
+      case None =>
+        RIO.fail(s"Source is a top level directory, can't copy. Source($path)")
+      case Some(filename) =>
+        val destLocalFile = destination.toLocalPath | filename
+        RIO.using(RIO.safe[InputStream](new FileInputStream(toFile)))(destLocalFile.writeStream(_)).as(
+          LocalFile.unsafe(destLocalFile.path.path))
+    }
+}
+
+object LocalFile {
+  /** Construct a path from a java.io.File. */
+  def fromFile(f: File): LocalFile =
```

Comment:
Should there be an `exists` call here, and should all of these constructors be in `RIO` given they have to be on disk.